### PR TITLE
[MGDOBR-860] Tests for updated webhook processor parameters

### PR DESCRIPTION
**Pull request by Aakanksha0311**
[MGDOBR-860](https://issues.redhat.com/browse/MGDOBR-860)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by sutaakar** [comment]
@Aakanksha0311 Please adjust the PR title to describe what is actually delivered, for example as JIRA title.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java [hunk 1]
```diff
@@ -21,21 +19,17 @@ public static List<WebhookSiteRequest> requests(WebhookSiteQuerySorting sorting)
                 .jsonPath().getList("data", WebhookSiteRequest.class);
     }
 
-    public static void deleteRequest(final WebhookSiteRequest request) {
+    public static void deleteRequest(final WebhookSiteRequest request, Object webhookID) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java:22]: IMHO `webhookID` can be declared as String, or is there some reason to declare it as Object?

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java [hunk 2]
```diff
@@ -65,13 +65,15 @@ public static void initializeLocalTestConfig() {
     public static void webhookSiteRequestHistoryIsCleared() {
         if (WebhookSiteResource.isSpecified()) {
             final LocalDate yesterday = LocalDate.now(ZoneId.systemDefault()).minusDays(1);
-            WebhookSiteResource.requests(WebhookSiteQuerySorting.OLDEST)
+            String webhookID = Utils.getSystemProperty("webhook.site.uuid");
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java:68]: Please declare this as a constant on top of the class.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java [hunk 3]
```diff
@@ -65,13 +65,15 @@ public static void initializeLocalTestConfig() {
...
+
+            WebhookSiteResource.requests(webhookID, WebhookSiteQuerySorting.OLDEST)
                     .stream()
                     .filter(request -> {
                         final LocalDate requestCreatedAt = request.getCreatedAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
                         return yesterday.isAfter(requestCreatedAt);
                     })
-                    .forEach(WebhookSiteResource::deleteRequest);
+                    .forEach(webhookID1 -> WebhookSiteResource.deleteRequest(webhookID1, webhookID));
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java:76]:
```suggestion
                    .forEach(request -> WebhookSiteResource.deleteRequest(request, webhookID));
```

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java [hunk 4]
```diff
@@ -26,27 +26,29 @@ public WebhookSiteSteps(TestContext context) {
         this.context = context;
     }
 
-    @Then("^Webhook site contains request with text \"(.*)\" within (\\d+) (?:minute|minutes)$")
-    public void webhookSiteContainsRequest(String requestText, int timeoutMinutes) {
+    @Then("^Webhook site with id \"(.*)\" contains request with text \"(.*)\" within (\\d+) (?:minute|minutes)$")
+    public void webhookSiteWithIdContainsRequest(String webhookID, String requestText, int timeoutMinutes) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java:30]:
```suggestion
    public void webhookSiteWithIdContainsRequest(String webhookId, String requestText, int timeoutMinutes) {
```
Also other occurrences below.

File: integration-tests/src/test/resources/features/action-webhook.feature [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
+  Background:
     Given authenticate against Manager
     And create a new Bridge "mybridge" in cloud provider "aws" and region "us-east-1"
     And the Bridge "mybridge" is existing with status "ready" within 4 minutes
     And the Ingress of Bridge "mybridge" is available within 2 minutes
 
+
+  Scenario: Webhook is correctly called
+
     And add a Processor to the Bridge "mybridge" with body:
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/action-webhook.feature:12]:
```suggestion
    Given add a Processor to the Bridge "mybridge" with body:
```

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java (hunk 1 above)
Comment (Aakanksha0311) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java:22]: Yes, I tried assigning it as a string but that doesn't work in this scenario not sure why, it gives error.

Comment (sutaakar) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java:22]: What error?

Comment (Aakanksha0311) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java:22]: In Hooks.java it doesn't accept the method parameter, when it is declared as a string, but when declared as Object it works fine.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java [hunk 6]
```diff
@@ -38,6 +38,7 @@
  */
 public class Hooks {
 
+    private static final String webhookId = Utils.getSystemProperty("webhook.site.uuid");
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java:41]: Please use capital letters for constant.

Comment (Aakanksha0311) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java:41]: Okay!

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java [hunk 7]
```diff
@@ -38,6 +38,7 @@
  */
 public class Hooks {
 
+    private static final String WEBHOOKID = Utils.getSystemProperty("webhook.site.uuid");
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java:41]: please use `WEBHOOK_ID` so the text is readable

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java [hunk 8]
```diff
@@ -2,17 +2,15 @@
...
 import io.restassured.RestAssured;
 
 public class WebhookSiteResource {
 
     private static final String ENDPOINT_BASE_URL = "https://webhook.site";
     private static final String ENDPOINT_UUID = "webhook.site.uuid";
 
-    public static List<WebhookSiteRequest> requests(WebhookSiteQuerySorting sorting) {
-        return RestAssured.get(ENDPOINT_BASE_URL + "/token/{webhookUuid}/requests?sorting={sorting}", getEndpointUuid(), sorting.getValue())
+    public static List<WebhookSiteRequest> requests(String webhookID, WebhookSiteQuerySorting sorting) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java:12]: please use webhookId

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java (hunk 7 above)
Comment (Aakanksha0311) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/Hooks.java]: Ok, Done.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java (hunk 8 above)
Comment (Aakanksha0311) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/webhook/site/WebhookSiteResource.java]: Ok

**Review by sutaakar — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]
LGTM. Thanks

**Pull request merged**